packages/client/src/request-login.ts

```diff
@@ -0,0 +1,24 @@
+import qs from 'query-string';
+
+import { generateCodeChallenge, generateCodeVerifier } from './generators';
+
+export const getLoginUrlAndCodeVerifier = (
+  baseUrl: string,
+  clientId: string,
+  redirectUri: string
+): [string, string] => {
+  const codeVerifier = generateCodeVerifier();
+  const codeChallenge = generateCodeChallenge(codeVerifier);
+
+  const url = `${baseUrl}?${qs.stringify({
+    client_id: clientId,
+    scope: 'openid offline_access',
+    response_type: 'code',
+    redirect_uri: [redirectUri],
+    prompt: 'consent',
+    code_challenge: codeChallenge,
+    code_challenge_method: 'S256',
+  })}`;
+
+  return [url, codeVerifier];
```

Comment:
Would defining the return type as a map better?  It might contain more informative keys on the return type rather than current [string, string]

Reply:
or we can write js docs for this func. i'm ok with both